pkgs/development/compilers/symbiflow-yosys/default.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenv
+, abc-verifier
+, bash
+, bison
+, fetchFromGitHub
+, flex
+, libffi
+, pkgconfig
+, protobuf
+, python3
+, readline
+, tcl
+, verilog
+, zlib
+, plugins ? []
+}:
+
+let
+  localAbc-verifier = abc-verifier.overrideAttrs (_: rec {
+    pname = "abc-verifier";
+    version = "2020.06.22";
+
+    src = fetchFromGitHub {
+      owner  = "YosysHQ";
+      repo   = "abc";
+      rev    = "341db25668f3054c87aa3372c794e180f629af5d";
+      sha256 = "14cgv34vz5ljkcms6nrv19vqws2hs8bgjgffk5q03cbxnm2jxv5s";
+    };
+
+    passthru.rev = src.rev;
+  });
+in
+
+stdenv.mkDerivation rec {
+  pname   = "symbiflow-yosys";
+  version = "0.8-6021-gd8b2d1a2";
+
+  src = fetchFromGitHub {
+    owner  = "SymbiFlow";
+    repo   = "yosys";
+    rev    = "d8b2d1a2b1a93057678cf49bb8f0329f191faba1";
+    sha256 = "1w8jnqzabvzixjllhb6ak2n2gmjvsn6qd996i7z70bsq5rgdkq9g";
+  };
+
+  enableParallelBuilding = true;
+  nativeBuildInputs = [ pkgconfig protobuf flex bison python3 ];
+  buildInputs = [ tcl readline libffi python3 protobuf zlib ];
+
+  cc = if stdenv.cc.isClang then "clang" else "gcc";
+
+  makeFlags = [
+    "LD=${stdenv.cc.targetPrefix}c++"
+    "ENABLE_PROTOBUF=1"
+    "PREFIX=${placeholder "out"}"
+  ];
+
+  postPatch = ''
+    substituteInPlace ./Makefile \
+      --replace 'echo UNKNOWN' 'echo ${builtins.substring 0 10 src.rev}'
+    chmod +x ./misc/yosys-config.in
+    patchShebangs ./misc/yosys-config.in
+    patchShebangs tests
+  '';
+
+  preBuild = let
+    shortAbcRev = builtins.substring 0 7 localAbc-verifier.rev;
+  in ''
+    chmod -R u+w .
+    make config-${cc}
+    echo 'ABCEXTERNAL = ${localAbc-verifier}/bin/abc' >> Makefile.conf
+
+    # we have to do this ourselves for some reason...
+    (cd misc && protoc --cpp_out ../backends/protobuf/ ./yosys.proto)
+
+    if ! grep -q "ABCREV = ${shortAbcRev}" Makefile; then
+      echo "yosys isn't compatible with the provided abc (${shortAbcRev}), failing."
+      exit 1
+    fi
+  '';
+
+  doCheck = true;
+  checkInputs = [ verilog ];
+
+  # Internally, yosys knows to use the specified hardcoded ABCEXTERNAL binary.
+  # But other tools (like mcy or symbiyosys) can't know how yosys was built, so
+  # they just assume that 'yosys-abc' is available -- but it's not installed
+  # when using ABCEXTERNAL
+  #
+  # add a symlink to fake things so that both variants work the same way.
+  postInstall = ''
+    ln -sfv ${localAbc-verifier}/bin/abc $out/bin/yosys-abc
+
+    mkdir -p $out/share/yosys/plugins
+    for plugin in $plugins; do
```

Comment:
I suppose you are missing `inherit plugins` somewhere

Reply:
It symlinks the plugins as expected, I don't think an inherit is necessary.

Reply:
I can't see how is that possible. The `plugins` argument defined in
https://github.com/NixOS/nixpkgs/blob/7c66a20bf43888332e6ad08af4107216408b6ab8/pkgs/development/compilers/symbiflow-yosys/default.nix#L15
is unused

Reply:
yeah, you're right, I managed to get the plugins symlinked because I was using overrideAttrs in my shell.nix